python/ray/tests/test_multi_node_3.py

```diff
@@ -90,6 +90,13 @@ def test_calling_start_ray_head(call_ray_stop_only):
         ["start", "--head", "--address", "127.0.0.1:6379", "--port", "0"])
     check_call_ray(["stop"])
 
+    # Test starting Ray with --external-kv-storage-config.
+    check_call_ray([
+        "start", "--head", "--external-kv-storage-config",
+        "redis:127.0.0.1:6379", "--port", "0"
```

Comment:
should we use `redis://` ?

Reply:
Yeah, I also think `redis://` is better. @mwtian

Reply:
Chatted with Yi Cheng this afternoon. I think the format we choose should support the following settings:
1. multiple Redis ip:port.
2. Redis password, together with (1) above.
3. non-redis backend in future, with potentially more complex configs.

With `redis://`, would it be `--external-kv-storage-config=redis://123:4?password=xxx,redis://124:5?password=yyy,redis://125:6?password=zzz`?

Reply:
what's in my mind is that if we are using redis:// kind of format, it'll be easier for future extension and also you can pass parameter in the URL and use urlparser in py to decode them.

Reply:
Hmm We need to support passing a separate password for each redis host. Currently only one same password used for multiple external redis.

Reply:
According to my understanding, the `--redis-password` option may be removed, if password not specified in cli the default password used.
e.g. For `--external-kv-storage-config=redis://host1:port1,redis://host2:port2?password=yyy`:
the default password used for host1, and `password=yyy` used for host2.

Reply:
Sounds good. Sorry for not being clear at the beginning. Ideally we can move all redis password and redis shard settings to be under this flag. Please let us know if you see other Redis configurations we should support here.

Reply:
Hmm After thinking it over, I doubt whether we really need to set multiple different passwords for multiple redis? I'm afraid in most of the time, users will set the same password for multiple redis. Maybe there be some over design here.
Currently, only one password required although multiple redis provided. https://github.com/ray-project/ray/pull/13170